### DISCUSSION TITLE
Questions regarding Graph Autoencoder and reconstructing input features + edge indices

### DISCUSSION THREAD

**Discussion by sivaramkrishnan101** [pyg-team/pytorch_geometric#7729] [Q&A]
Hello everyone,

**Problem** - I'm currently working on implementing a Graph Autoencoder using a custom dataset. 

**Scenario** - The dataset consists of 20 graphs, each with 5 nodes and 2 features. These graphs are fully connected, and I have generated node embeddings using the SAGECONV method. z = SAGECONV(2, 2) -> SAGECONV(2, 1)

**Questions** - How can I define a decoder architecture to 

- Reconstruct the input matrix (I am thinking of using the FCN layers for this)?  
- Reconstruct the adjacency matrix.

I hope to get some ideas on this implementation as I have not seen a GAE defined to reconstruct the input features so far. Thank you :)

**Comment by EdisonLeeeee** [comment]
You might find the following repositories helpful:
+ Reconstruct features: https://github.com/THUDM/GraphMAE
+ Reconstruct structure: https://github.com/EdisonLeeeee/MaskGAE

**Reply by sivaramkrishnan101** [reply]
Hi Edison,

I appreciate your response. This is exactly what I was looking for. Thank you very much!

**Reply by sujoyrc** [reply]
what if I want to recover both with a weighted loss between the two components. My node features are text and I can embed the text via a SentenceTransformer model into an embedding. every node has only 1 text
